packages/vats/test/upgrading/test-upgrade-vats.js

```diff
@@ -229,6 +239,178 @@ test('upgrade vat-bridge', async t => {
+  ]);
+
+  t.log(`create a non-bridged bank manager`);
+  /** @type {ERef<BankManager>} */
+  const noBridgeMgr = await EV.vat('bank').makeBankManager();
+
+  t.log(`create a bridged bank manager`);
+  const dev = await EV.vat('bootstrap').getDevice('bridge');
+  const bridge1 = await EV.vat('bridge').provideManagerForBridge(dev);
+  const bankBridge = await EV(bridge1).register(BridgeId.BANK);
+  /** @type {ERef<BankManager>} */
+  const bridgedMgr = await EV.vat('bank').makeBankManager(bankBridge);
+
+  t.log('subscribe to no bridge asset lists');
+  const nbit = await EV(await EV(noBridgeMgr).getAssetSubscription())[
+    Symbol.asyncIterator
+  ]();
+
+  t.log('add an asset to both');
+  const abcKit = await EV.vat('mint').makeIssuerKit('ABC');
+  const otherKit = await EV.vat('mint').makeIssuerKit('OTHER');
+  await EV(noBridgeMgr).addAsset('uabc', 'ABC', 'A Bank Coin', abcKit);
+  await EV(bridgedMgr).addAsset('uabc', 'ABC', 'A Bank Coin', abcKit);
+  await EV(bridgedMgr).addAsset('uother', 'OTHER', 'Other Coin', otherKit);
+
+  const banks = {
+    agoric1234: await EV(noBridgeMgr).getBankForAddress('agoric1234'),
+    agoric1bbb: await EV(bridgedMgr).getBankForAddress('agoric1bbb'),
+  };
+
+  const abcPurse1 = await EV(banks.agoric1234).getPurse(abcKit.brand);
+  const abcPurse2 = await EV(banks.agoric1234).getPurse(abcKit.brand);
+  t.is(abcPurse1, abcPurse2, 'must get the same purse back');
+
+  const otherPurse1 = await EV(banks.agoric1bbb).getPurse(otherKit.brand);
+  const otherPurse2 = await EV(banks.agoric1bbb).getPurse(otherKit.brand);
+  t.is(otherPurse1, otherPurse2, 'must get the same purse back');
+
+  const pmt1 = await EV(abcKit.mint).mintPayment({
+    brand: abcKit.brand,
+    value: 1000n,
+  });
+  await EV(abcPurse1).deposit(pmt1);
+  t.deepEqual(
+    await EV(abcPurse1).getCurrentAmount(),
+    { brand: abcKit.brand, value: 1000n },
+    'purse must have 1000n',
+  );
+
+  const abcAsset = harden({
+    brand: abcKit.brand,
+    denom: 'uabc',
+    issuer: abcKit.issuer,
+    issuerName: 'ABC',
+    proposedName: 'A Bank Coin',
+  });
+  const otherAsset = harden({
+    brand: otherKit.brand,
+    denom: 'uother',
+    issuer: otherKit.issuer,
+    issuerName: 'OTHER',
+    proposedName: 'Other Coin',
+  });
+
+  t.log('subscribe to bridged asset list');
+  const bit = await EV(await EV(bridgedMgr).getAssetSubscription())[
+    Symbol.asyncIterator
+  ]();
+
+  t.deepEqual(await EV(bit).next(), { value: abcAsset, done: false });
+
+  t.log('now perform the null upgrade');
+  const { incarnationNumber } = await EV.vat('bootstrap').upgradeVat(
+    bankVatConfig,
+  );
+
+  t.is(incarnationNumber, 1, 'Bank vat must be upgraded');
+
+  await t.throwsAsync(() => EV(nbit).next(), { message: 'vat terminated' });
+  await t.throwsAsync(() => EV(bit).next(), { message: 'vat terminated' });
+
+  const abcPurse3 = await EV(banks.agoric1234).getPurse(abcKit.brand);
+  t.is(abcPurse3, abcPurse1, 'must get the same purse back');
```

Comment:
please also test the purse balances. it's possible to have a durable object that returns values from heap.